// @aws-sdk/client-cloudwatch/dist-cjs/CloudWatch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudWatch = void 0;
const CloudWatchClient_1 = require("./CloudWatchClient");
const DeleteAlarmsCommand_1 = require("./commands/DeleteAlarmsCommand");
const DeleteAnomalyDetectorCommand_1 = require("./commands/DeleteAnomalyDetectorCommand");
const DeleteDashboardsCommand_1 = require("./commands/DeleteDashboardsCommand");
const DeleteInsightRulesCommand_1 = require("./commands/DeleteInsightRulesCommand");
const DeleteMetricStreamCommand_1 = require("./commands/DeleteMetricStreamCommand");
const DescribeAlarmHistoryCommand_1 = require("./commands/DescribeAlarmHistoryCommand");
const DescribeAlarmsCommand_1 = require("./commands/DescribeAlarmsCommand");
const DescribeAlarmsForMetricCommand_1 = require("./commands/DescribeAlarmsForMetricCommand");
const DescribeAnomalyDetectorsCommand_1 = require("./commands/DescribeAnomalyDetectorsCommand");
const DescribeInsightRulesCommand_1 = require("./commands/DescribeInsightRulesCommand");
const DisableAlarmActionsCommand_1 = require("./commands/DisableAlarmActionsCommand");
const DisableInsightRulesCommand_1 = require("./commands/DisableInsightRulesCommand");
const EnableAlarmActionsCommand_1 = require("./commands/EnableAlarmActionsCommand");
const EnableInsightRulesCommand_1 = require("./commands/EnableInsightRulesCommand");
const GetDashboardCommand_1 = require("./commands/GetDashboardCommand");
const GetInsightRuleReportCommand_1 = require("./commands/GetInsightRuleReportCommand");
const GetMetricDataCommand_1 = require("./commands/GetMetricDataCommand");
const GetMetricStatisticsCommand_1 = require("./commands/GetMetricStatisticsCommand");
const GetMetricStreamCommand_1 = require("./commands/GetMetricStreamCommand");
const GetMetricWidgetImageCommand_1 = require("./commands/GetMetricWidgetImageCommand");
const ListDashboardsCommand_1 = require("./commands/ListDashboardsCommand");
const ListManagedInsightRulesCommand_1 = require("./commands/ListManagedInsightRulesCommand");
const ListMetricsCommand_1 = require("./commands/ListMetricsCommand");
const ListMetricStreamsCommand_1 = require("./commands/ListMetricStreamsCommand");
const ListTagsForResourceCommand_1 = require("./commands/ListTagsForResourceCommand");
const PutAnomalyDetectorCommand_1 = require("./commands/PutAnomalyDetectorCommand");
const PutCompositeAlarmCommand_1 = require("./commands/PutCompositeAlarmCommand");
const PutDashboardCommand_1 = require("./commands/PutDashboardCommand");
const PutInsightRuleCommand_1 = require("./commands/PutInsightRuleCommand");
const PutManagedInsightRulesCommand_1 = require("./commands/PutManagedInsightRulesCommand");
const PutMetricAlarmCommand_1 = require("./commands/PutMetricAlarmCommand");
const PutMetricDataCommand_1 = require("./commands/PutMetricDataCommand");
const PutMetricStreamCommand_1 = require("./commands/PutMetricStreamCommand");
const SetAlarmStateCommand_1 = require("./commands/SetAlarmStateCommand");
const StartMetricStreamsCommand_1 = require("./commands/StartMetricStreamsCommand");
const StopMetricStreamsCommand_1 = require("./commands/StopMetricStreamsCommand");
const TagResourceCommand_1 = require("./commands/TagResourceCommand");
const UntagResourceCommand_1 = require("./commands/UntagResourceCommand");
class CloudWatch extends CloudWatchClient_1.CloudWatchClient {
    deleteAlarms(args, optionsOrCb, cb) {
        const command = new DeleteAlarmsCommand_1.DeleteAlarmsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteAnomalyDetector(args, optionsOrCb, cb) {
        const command = new DeleteAnomalyDetectorCommand_1.DeleteAnomalyDetectorCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteDashboards(args, optionsOrCb, cb) {
        const command = new DeleteDashboardsCommand_1.DeleteDashboardsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteInsightRules(args, optionsOrCb, cb) {
        const command = new DeleteInsightRulesCommand_1.DeleteInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    deleteMetricStream(args, optionsOrCb, cb) {
        const command = new DeleteMetricStreamCommand_1.DeleteMetricStreamCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeAlarmHistory(args, optionsOrCb, cb) {
        const command = new DescribeAlarmHistoryCommand_1.DescribeAlarmHistoryCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeAlarms(args, optionsOrCb, cb) {
        const command = new DescribeAlarmsCommand_1.DescribeAlarmsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeAlarmsForMetric(args, optionsOrCb, cb) {
        const command = new DescribeAlarmsForMetricCommand_1.DescribeAlarmsForMetricCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeAnomalyDetectors(args, optionsOrCb, cb) {
        const command = new DescribeAnomalyDetectorsCommand_1.DescribeAnomalyDetectorsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    describeInsightRules(args, optionsOrCb, cb) {
        const command = new DescribeInsightRulesCommand_1.DescribeInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    disableAlarmActions(args, optionsOrCb, cb) {
        const command = new DisableAlarmActionsCommand_1.DisableAlarmActionsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    disableInsightRules(args, optionsOrCb, cb) {
        const command = new DisableInsightRulesCommand_1.DisableInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    enableAlarmActions(args, optionsOrCb, cb) {
        const command = new EnableAlarmActionsCommand_1.EnableAlarmActionsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    enableInsightRules(args, optionsOrCb, cb) {
        const command = new EnableInsightRulesCommand_1.EnableInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getDashboard(args, optionsOrCb, cb) {
        const command = new GetDashboardCommand_1.GetDashboardCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getInsightRuleReport(args, optionsOrCb, cb) {
        const command = new GetInsightRuleReportCommand_1.GetInsightRuleReportCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getMetricData(args, optionsOrCb, cb) {
        const command = new GetMetricDataCommand_1.GetMetricDataCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getMetricStatistics(args, optionsOrCb, cb) {
        const command = new GetMetricStatisticsCommand_1.GetMetricStatisticsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getMetricStream(args, optionsOrCb, cb) {
        const command = new GetMetricStreamCommand_1.GetMetricStreamCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    getMetricWidgetImage(args, optionsOrCb, cb) {
        const command = new GetMetricWidgetImageCommand_1.GetMetricWidgetImageCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listDashboards(args, optionsOrCb, cb) {
        const command = new ListDashboardsCommand_1.ListDashboardsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listManagedInsightRules(args, optionsOrCb, cb) {
        const command = new ListManagedInsightRulesCommand_1.ListManagedInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listMetrics(args, optionsOrCb, cb) {
        const command = new ListMetricsCommand_1.ListMetricsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listMetricStreams(args, optionsOrCb, cb) {
        const command = new ListMetricStreamsCommand_1.ListMetricStreamsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    listTagsForResource(args, optionsOrCb, cb) {
        const command = new ListTagsForResourceCommand_1.ListTagsForResourceCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putAnomalyDetector(args, optionsOrCb, cb) {
        const command = new PutAnomalyDetectorCommand_1.PutAnomalyDetectorCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putCompositeAlarm(args, optionsOrCb, cb) {
        const command = new PutCompositeAlarmCommand_1.PutCompositeAlarmCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putDashboard(args, optionsOrCb, cb) {
        const command = new PutDashboardCommand_1.PutDashboardCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putInsightRule(args, optionsOrCb, cb) {
        const command = new PutInsightRuleCommand_1.PutInsightRuleCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putManagedInsightRules(args, optionsOrCb, cb) {
        const command = new PutManagedInsightRulesCommand_1.PutManagedInsightRulesCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putMetricAlarm(args, optionsOrCb, cb) {
        const command = new PutMetricAlarmCommand_1.PutMetricAlarmCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putMetricData(args, optionsOrCb, cb) {
        const command = new PutMetricDataCommand_1.PutMetricDataCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    putMetricStream(args, optionsOrCb, cb) {
        const command = new PutMetricStreamCommand_1.PutMetricStreamCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    setAlarmState(args, optionsOrCb, cb) {
        const command = new SetAlarmStateCommand_1.SetAlarmStateCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    startMetricStreams(args, optionsOrCb, cb) {
        const command = new StartMetricStreamsCommand_1.StartMetricStreamsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    stopMetricStreams(args, optionsOrCb, cb) {
        const command = new StopMetricStreamsCommand_1.StopMetricStreamsCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    tagResource(args, optionsOrCb, cb) {
        const command = new TagResourceCommand_1.TagResourceCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
    untagResource(args, optionsOrCb, cb) {
        const command = new UntagResourceCommand_1.UntagResourceCommand(args);
        if (typeof optionsOrCb === "function") {
            this.send(command, optionsOrCb);
        }
        else if (typeof cb === "function") {
            if (typeof optionsOrCb !== "object")
                throw new Error(`Expect http options but get ${typeof optionsOrCb}`);
            this.send(command, optionsOrCb || {}, cb);
        }
        else {
            return this.send(command, optionsOrCb);
        }
    }
}
exports.CloudWatch = CloudWatch;
