// @aws-solutions-constructs/core/lib/cognito-defaults.js
"use strict";
/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultUserPoolProps = exports.DefaultUserPoolClientProps = exports.DefaultIdentityPoolProps = void 0;
const DefaultUserPoolProps = {};
exports.DefaultUserPoolProps = DefaultUserPoolProps;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultIdentityPoolProps(userPoolClientId, userPoolProviderName) {
    return {
        allowUnauthenticatedIdentities: false,
        cognitoIdentityProviders: [{
                clientId: userPoolClientId,
                providerName: userPoolProviderName,
                serverSideTokenCheck: true
            }]
    };
}
exports.DefaultIdentityPoolProps = DefaultIdentityPoolProps;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultUserPoolClientProps(userpool) {
    return {
        userPool: userpool
    };
}
exports.DefaultUserPoolClientProps = DefaultUserPoolClientProps;
