// @aws-solutions-constructs/core/lib/elasticache-defaults.js
"use strict";
/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetMemcachedDefaults = exports.GetDefaultCachePort = void 0;
/*
 *  The functions found here in the core library are for internal use and can be changed
 *  or removed outside of a major release. We recommend against calling them directly from client code.
 */
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function GetDefaultCachePort() {
    // Best practice not to use default port 11211
    return 11222;
}
exports.GetDefaultCachePort = GetDefaultCachePort;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function GetMemcachedDefaults(id, port) {
    return {
        clusterName: `${id}-cdk-cluster`,
        cacheNodeType: "cache.t3.medium",
        engine: "memcached",
        numCacheNodes: 2,
        port,
        azMode: 'cross-az'
    };
}
exports.GetMemcachedDefaults = GetMemcachedDefaults;
