// @aws-solutions-constructs/core/lib/iot-topic-rule-defaults.js
"use strict";
/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultCfnTopicRuleProps = void 0;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultCfnTopicRuleProps(_actions, _sql = '') {
    const _topicRulePayload = {
        ruleDisabled: false,
        actions: _actions,
        sql: _sql
    };
    const defaultCfnTopicRuleProps = {
        topicRulePayload: _topicRulePayload
    };
    return defaultCfnTopicRuleProps;
}
exports.DefaultCfnTopicRuleProps = DefaultCfnTopicRuleProps;
