// @aws-solutions-constructs/core/lib/sagemaker-defaults.js
"use strict";
/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultSagemakerEndpointProps = exports.DefaultSagemakerEndpointConfigProps = exports.DefaultSagemakerModelProps = exports.DefaultSagemakerNotebookProps = void 0;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultSagemakerNotebookProps(roleArn, kmsKeyId, subnetId, securityGroupIds) {
    return {
        instanceType: 'ml.t2.medium',
        roleArn,
        kmsKeyId,
        ...(subnetId && { subnetId, directInternetAccess: 'Disabled' }),
        ...(securityGroupIds && { securityGroupIds }),
    };
}
exports.DefaultSagemakerNotebookProps = DefaultSagemakerNotebookProps;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultSagemakerModelProps(executionRoleArn, primaryContainer, vpcConfig) {
    return {
        executionRoleArn,
        primaryContainer,
        ...(vpcConfig && { vpcConfig }),
    };
}
exports.DefaultSagemakerModelProps = DefaultSagemakerModelProps;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultSagemakerEndpointConfigProps(modelName, kmsKeyId) {
    return {
        productionVariants: [
            {
                modelName,
                initialInstanceCount: 1,
                initialVariantWeight: 1.0,
                instanceType: 'ml.m4.xlarge',
                variantName: 'AllTraffic',
            },
        ],
        ...(kmsKeyId && { kmsKeyId }),
    };
}
exports.DefaultSagemakerEndpointConfigProps = DefaultSagemakerEndpointConfigProps;
/**
 * @internal This is an internal core function and should not be called directly by Solutions Constructs clients.
 */
function DefaultSagemakerEndpointProps(endpointConfigName) {
    return {
        endpointConfigName,
    };
}
exports.DefaultSagemakerEndpointProps = DefaultSagemakerEndpointProps;
